frontend/ui/widget/numberpickerwidget.lua

```diff
@@ -142,14 +157,32 @@ function NumberPickerWidget:changeValue(value, step, max, min)
     else
         value = value + step
         if value > max then
-            value = value - max + 1
+            value = min
         elseif value < min then
-            value = max + 1 + value
+            value = max
         end
     end
     return value
 end
 
+function NumberPickerWidget:getDaysInMonth(month, year)
+    local days_in_month = { 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 }
+    local days = days_in_month[month]
+    -- check for leap year
```

Comment:
Hah, you made me look back at something I wrote over 14 years ago ([link](http://fransdejonge.com/calendar/)).

```javascript
// makes sure months have the right length...
var leapdays = new Array(31,29,31, 30,31,30, 31,31,30, 31,30,31); 
var yeardays = new Array(31,28,31, 30,31,30, 31,31,30, 31,30,31); 

// […]

function isLeapYear( year ) {
  // is it leap year ? returns a boolean
  return ( (0 == (year%4)) && ( (0 != (year%100)) || (0 == (year%400))));
  // ie, if the year divides by 4, but not by 100 except when it divides by
  // 400, it is leap year
}
```

Reply:
;-)